packages/nuxt/src/components/templates.ts

```diff
@@ -118,20 +118,27 @@ export const componentsTypeTemplate = {
     })
 
     const islandType = 'type IslandComponent<T extends DefineComponent> = T & DefineComponent<{}, {refresh: () => Promise<void>}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, SlotsType<{ fallback: { error: unknown } }>>'
+    const components: string[] = new Array(componentTypes.length)
+    const componentExports: string[] = new Array(componentTypes.length)
+    for (let i = 0; i < componentTypes.length; i++) {
+      const [pascalName, type] = componentTypes[i]!
+      components[i] = `${pascalName}': ${type}`
+      componentExports[i] = `${pascalName}: ${type}`
+    }
     return `
 import type { DefineComponent, SlotsType } from 'vue'
 ${nuxt.options.experimental.componentIslands ? islandType : ''}
 interface _GlobalComponents {
-  ${componentTypes.map(([pascalName, type]) => `    '${pascalName}': ${type}`).join('\n')}
-  ${componentTypes.map(([pascalName, type]) => `    'Lazy${pascalName}': ${type}`).join('\n')}
+    '${components.join('\n    \'')}
+    'Lazy${components.join('\n    \'Lazy')}
```

Comment:
I'm pretty sure there's no benefit to switching from `.map` as this already is able to optimise by creating an array with the correct number of elements.

Reply:
That's exactly the problem, it makes 4 total arrays and 4 total iterations.

Let's take the Vuetify Nuxt module for example, given that they're working on migrating from a custom Vite plugin to Nuxt's component system:

They have, from a rough estimate of mine, 76 components.

The current approach makes a total of 304 elements in 4 arrays (76 elements in each array) and 4 iterations over them (so 304 counts in total).

The new approach makes a total of 152 elements in 2 arrays (76 elements in each array) and 1 iteration only (so 76 counts in total).

All of this calculation does not even take into consideration the delayed hydration PR, which will introduce LazyIdle, LazyVisible, LazyPromise, LazyTime, LazyMedia, LazyEvent, and  LazyNever, which will make it 9 types in total (76 * 9 = 684 components in total)